proto/identity/associations/association.proto

```diff
@@ -0,0 +1,87 @@
+// Payloads to be signed for identity associations
+syntax = "proto3";
+
+package xmtp.identity.associations;
+
+import "identity/associations/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/identity/associations";
+option java_package = "org.xmtp.proto.identity.associations";
+
+// The first entry of any XID log. The XID must be deterministically derivable
+// from the address and nonce.
+// The recovery address defaults to the initial associated_address unless
+// there is a subsequent ChangeRecoveryAddress in the log.
+message CreateXid {
+  string xid = 1;
+  uint32 nonce = 2;
+  Signature initial_address_signature = 3;  // Must be an addressable member
+}
+
+// Adds a new member for an XID - either an addressable member such as a
+// wallet, or an installation acting on behalf of an address.
+// A key-pair that has been associated with one role MUST not be permitted to be
+// associated with a different role.
+message AddAssociation {
+  // The role of the new member in the association. Must not be UNSPECIFIED.
+  enum AssociatedRole {
+    ASSOCIATED_ROLE_UNSPECIFIED = 0;
+    ASSOCIATED_ROLE_INSTALLATION = 1;
+    ASSOCIATED_ROLE_ADDRESS = 2;
+  }
+
+  string xid = 1;
+  uint64 client_timestamp_ns = 2;
+  AssociatedRole new_member_role = 3;
+  Signature existing_member_signature = 4;
+  Signature new_member_signature = 5;
```

Comment:
There's a circular dependency here with generating the signature text. 

We need the new member's address or public key to generate the text, but we need the text to recover an identity out of most of these signature types.

Reply:
The easy fix is to at least include the identifier for the new entity. Maybe the existing one too